pczt/src/roles/updater/sapling.rs

```diff
@@ -0,0 +1,38 @@
+use sapling::pczt::{ParseError, Updater, UpdaterError};
+
+use crate::Pczt;
+
+impl super::Updater {
+    /// Updates the Sapling bundle with information in the given closure.
+    pub fn update_sapling_with<F>(self, f: F) -> Result<Self, SaplingError>
+    where
+        F: FnOnce(Updater<'_>) -> Result<(), UpdaterError>,
+    {
+        let Pczt {
+            global,
+            transparent,
+            sapling,
+            orchard,
+        } = self.pczt;
+
+        let mut bundle = sapling.into_parsed().map_err(SaplingError::Parser)?;
+
+        bundle.update_with(f).map_err(SaplingError::Updater)?;
```

Comment:
This doesn't check the modifiability flags.

Reply:
Those are irrelevant here, as the only permitted changes via an Updater are non-effecting.